### PR TITLE
added pawn promotion methods and move_piece_onscreen pusher trigger

### PR DESCRIPTION
**Pull request by FsDevNinja**
Here we are showing the user piece options if they get a pawn to the otherside.

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -67,7 +72,7 @@ def destroy
   end
   
   def add_player
-    current_player.join_game!(@game)
+    current_player.join_game!(@game, current_player)
```
Comment (kellylougheed) [inline app/controllers/games_controller.rb]: This is throwing an error for me because the join_game! method is expecting only one argument.

Comment (FsDevNinja) [inline app/controllers/games_controller.rb]: shoot.. that should have been removed.

File: app/assets/javascripts/games.js.coffee [hunk 2]
```diff
@@ -28,3 +28,18 @@ $ ->
...
+    $('#Queen-promotion').css('display', 'inline')
+    $('#Rook-promotion').css('display', 'inline')
+    $('#Knight-promotion').css('display', 'inline')
+    $('#Bishop-promotion').css('display', 'inline')
+
+  window.private_channel.bind 'pawn-promotion-choosen', (event) ->
+    $('#Queen-promotion').css('display', 'none')
+    $('#Rook-promotion').css('display', 'none')
+    $('#Knight-promotion').css('display', 'none')
+    $('#Bishop-promotion').css('display', 'none')
```
Comment (kellylougheed) [inline app/assets/javascripts/games.js.coffee]: Hmmm... these buttons didn't seem to disappear when I tested it!

Comment (kellylougheed) [inline app/assets/javascripts/games.js.coffee]: It doesn't look like this event is invoked anywhere.

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -2,12 +2,17 @@ class PiecesController < ApplicationController
...
     if selected_piece.move_to(params[:piece][:x_position].to_i,params[:piece][:y_position].to_i)
-      if selected_piece.game.has_started == false
-        selected_piece.game.has_started = true
-        selected_piece.game.save
-        Pusher["broadcast"].trigger!('game_has_started',{game_has_started: true})
-      end
-      
+        if selected_piece.can_pawn_promote?(params[:piece][:y_position].to_i)
+           Pusher["private-user_#{current_player.id}"].trigger!('pawn-promotion', {
+          pawnPromotionUpdateURL: game_piece_promote_pawn_path(@game, selected_piece.id), message: "You made it to the other side! Click one of the buttons to pick your new piece."
```
Comment (kellylougheed) [inline app/controllers/pieces_controller.rb]: I wonder if this is the line which is making it possible for me to promote white pawns in my current game, but not black pawns. When we're updating a piece, we want to notify the channel of whoever owns the piece, not necessarily the current_player.

File: app/assets/javascripts/games.js.coffee (hunk 2 above)
Comment (FsDevNinja) [inline app/assets/javascripts/games.js.coffee]: yeah.. it wasn't being triggered in the pawn-promotion method... just added it..

File: app/assets/javascripts/games.js.coffee [hunk 4]
```diff
@@ -28,3 +28,18 @@ $ ->
 
   window.private_channel.bind 'show_forfeit_button', (event) ->
     $('.forfeit').css('display', 'inline')
+
+  window.private_channel.bind 'pawn-promotion', (event) ->
+    window.pawnPromotionUpdateURL = event.pawnPromotionUpdateURL
+    $('#message').css('display', 'block')
+    $('#message').html(event.message)
+    $('#Queen-promotion').css('display', 'inline')
```
Comment (lasley) [inline app/assets/javascripts/games.js.coffee]: Naming pattern for IDs in HTML is lowerCamelCase - `queenPromotion`

File: app/assets/javascripts/pawnPromotion.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+$(document).ready(function() {
+    $("#Queen-promotion").click(function () {
```
Comment (lasley) [inline app/assets/javascripts/pawnPromotion.js]: Lots of duplication here. Why not make a `piece-promotion` class, then ` data-type` with the piece type. Then just use `piece.data('type')` for the `new_piece` declaration

File: app/views/games/show.html.erb [hunk 6]
```diff
@@ -47,7 +47,7 @@
 <div class="game-wrapper" data-current-turn="<%= @game.current_player_turn %>" data-player="<%= get_player %>" data-black-time-left="<%= @black_player_timer.time_left %>" data-white-time-left="<%= @white_player_timer.time_left %>" data-game-id="<%= @game.id %>" data-time-now="<%= Time.now.to_i %>" data-white-start-time="<%= @white_player_timer.start_time.to_i %>" data-black-start-time="<%= @black_player_timer.start_time.to_i %>" data-game-type="<%= @game.is_blitz %>" data-has-started="<%= @game.has_started %>">
   <div class="white-player">White Player's Time Left:<div id="white-player-timer"></div></div>
   <% else %>
-  <div class="game-wrapper" data-current-turn="<%= @game.current_player_turn %>">
+  <div class="game-wrapper" data-current-turn="<%= @game.current_player_turn %>" data-has-started="<%= @game.has_started %>>
```
Comment (kellylougheed) [inline app/views/games/show.html.erb]: I think you're missing a quote right before the ending >. (Just taking this fix for my branch as well, thanks!)

**Pull request merged**